src/compiler/scala/tools/nsc/io/SourceReader.scala

```diff
@@ -38,7 +39,11 @@ class SourceReader(decoder: CharsetDecoder, reporter: Reporter) {
     val c = new FileInputStream(file).getChannel
 
     try read(c)
-    catch { case e: Exception => reportEncodingError("" + file) ; Array() }
+    catch {
+      case ex: InterruptedException => throw ex
+      case _: ClosedByInterruptException => throw new InterruptedException
+      case e: Exception => reportEncodingError("" + file) ; Array()
```

Comment:
Instead of adding this pattern in various places, could we just change `case _: Throwable` (or `case _: Exception`) to `case NonFatal(_)`? Should `ClosedByInterruptException` be added to the fatal list in `NonFatal`?

Reply:
Maybe, and maybe :)

But I'm not really game to change `NonFatal` itself (that could change user code, for the better or worse), and I'm also reluctant to make transition the compiler to using `NonFatal` in a piecemeal manner (we should do this refactoring holisitcally with some design and tests)